app-header-layout/app-header-layout.html

```diff
@@ -126,7 +126,9 @@
           value: false,
           reflectToAttribute: true,
           observer: '_hasScrollingRegionChanged'
-        }
+        },
+
+        _savedPageYOffset: Number
```

Comment:
I'd suggest renaming this property as `_scrollTop: 0` and taking it out the property block. wdyt?